kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -284,9 +285,25 @@ impl<'tcx> GotocCtx<'tcx> {
                             TagEncoding::Direct => {
                                 // then the scalar field stores the discriminant
                                 let discr_ty = self.codegen_enum_discr_typ(ty);
-
                                 let init = self.codegen_scalar(s, discr_ty, span);
-                                self.codegen_direct_literal(ty, init)
+                                let cgt = self.codegen_ty(ty);
+                                let fields =
+                                    cgt.get_non_empty_components(&self.symbol_table).unwrap();
+                                assert_eq!(
```

Comment:
Please add a comment explaining "why" this assert is here.

Reply:
added